src/multihash.cr

```diff
@@ -2,29 +2,75 @@
 require "./hashes"
 
 module Multihash
-  class Multihash
-    @function_code : Int32
+  HEX_SIZE_IN_BYTES = 2
+  struct Multihash
+    @hash_code : Int32
+    @len : Int32
 
-    def initialize(hash_function : String, len : Int32, digest : String)
+    def initialize(hash_name : String, digest : String)
```

Comment:
We could also have an initializer with `digest` as `Bytes`:
```crystal
def initialize(hash_name : String, digest : Bytes)
```
The `Bytes` type is an alias for `Slice(UInt8)`, which has a `hexstring` method that converts it to a hex string. Or, maybe we keep the digest internally represented as `Bytes`, and fix this method. I'm not sure which makes more sense here.